tests/e2e/developer_pages.py

```diff
@@ -276,6 +281,119 @@ def click_add_question(self) -> SelectQuestionTypePage:
 
+    def click_edit_question(self, question_name: str) -> "EditQuestionPage":
+        self.page.get_by_role("link", name=question_name).click()
+        edit_question_page = EditQuestionPage(
+            self.page,
+            self.domain,
+            grant_name=self.grant_name,
+            collection_name=self.collection_name,
+            section_title=self.section_title,
+            form_name=self.form_name,
+        )
+        expect(edit_question_page.heading).to_be_visible()
+        return edit_question_page
+
+
+class EditQuestionPage(GrantDevelopersBasePage):
+    add_validation_button: Locator
+    back_link: Locator
+
+    def __init__(
+        self, page: Page, domain: str, grant_name: str, collection_name: str, section_title: str, form_name: str
+    ) -> None:
+        super().__init__(
+            page,
+            domain,
+            grant_name=grant_name,
+            heading=page.get_by_role("heading", name="Edit question"),
+        )
+        self.section_title = section_title
+        self.collection_name = collection_name
+        self.form_name = form_name
+        self.add_validation_button = self.page.get_by_role("button", name="Add validation").or_(
+            self.page.get_by_role("button", name="Add more validation")
+        )
+        self.back_link = self.page.get_by_role("link", name="Back")
+
+    def click_add_validation(self) -> "AddValidationPage":
+        self.add_validation_button.click()
+        add_validation_page = AddValidationPage(
+            self.page,
+            self.domain,
+            grant_name=self.grant_name,
+            collection_name=self.collection_name,
+            section_title=self.section_title,
+            form_name=self.form_name,
+        )
+        expect(add_validation_page.heading).to_be_visible()
+        return add_validation_page
+
+    def click_back(self) -> "ManageFormPage":
+        self.back_link.click()
+        manage_form_page = ManageFormPage(
+            self.page,
+            self.domain,
+            grant_name=self.grant_name,
+            collection_name=self.collection_name,
+            section_title=self.section_title,
+            form_name=self.form_name,
+        )
+        expect(manage_form_page.heading).to_be_visible()
+        return manage_form_page
+
+
+class AddValidationPage(GrantDevelopersBasePage):
+    add_validation_button: Locator
+
+    def __init__(
+        self, page: Page, domain: str, grant_name: str, collection_name: str, section_title: str, form_name: str
+    ) -> None:
+        super().__init__(
+            page,
+            domain,
+            grant_name=grant_name,
+            heading=page.get_by_role("heading", name="Add validation"),
+        )
+        self.section_title = section_title
+        self.collection_name = collection_name
+        self.form_name = form_name
+        self.add_validation_button = self.page.get_by_role("button", name="Add validation")
```

Comment:
The smallest of nits but I wonder if want to standardise on having like a `submit_button` for pages with a final submission action - it took me more brain cycles than I care to admit to realise that one `add_validation_button` was a link to this page and then another `add_validation_button` but a submission action (but that is likely a me problem)